deploy/dynamo/sdk/src/dynamo/sdk/cli/serving.py

```diff
@@ -411,8 +442,14 @@ def serve_http(
                 except json.JSONDecodeError as e:
                     logger.warning(f"Failed to parse DYNAMO_SERVICE_ENVS: {e}")
 
+            # HACK: WAR issues with running MPI_Spawn in TRTLLM internals
+            #       while in a Slurm environment.
+            # FIXME: Find a better solution for running TRTLLM workers with `dynamo serve`
+            cmd, dynamo_args = maybe_get_slurm_mpirun_command(svc.name, dynamo_args)
```

Comment:
An interesting data point here @tanmayv25 -

When killing the main process that launches a graph, ex: `dynamo serve graphs/agg.py ...` -- it takes the slurm job process down with it as we know.

However, when killing the standalone component process in this section of the code (PrefillWorker in this example), it gracefully shuts down and doesn't kill the slurm job with it. Sharing in case that helps us investigate the shutdown issues further.

Reply:
I wasn't expecting this. But interesting finding. Did you try `mpirun dynamo serve graphs/agg.py ...` as well for the first?

Reply:
I tried it, it didn't work. I don't have the error handy, but TLDR I don't think the MPI settings propagate correctly to subprocesses so the MPI comm spawn will still fail. I can probably get the actual error message later for future reference